### DISCUSSION TITLE
Are out-of-order moves allowed?

### DISCUSSION THREAD

**Discussion by KozhevnikovIvanA** [boardgameio/boardgame.io#1065] [Q&A]
Hey folks!
I am working on one game, where people play cards during their move. But **other** person can 'interupt' the move by playing a special card.

Imagine that 'Player A' wants to play cards A, B and C. And ideally they play all of them before they end turn. However 'Player B' has a special card D, which allowes to interupt 'Player A' turn and play card D which will affect game state.

Is there a good way to do it in boardgame.io framework?

**Comment by delucis** [comment]
Hi @KozhevnikovIvanA! 👋 

I think the feature you're looking for is what boardgame.io calls “Stages”. Here's the documentation page: https://boardgame.io/documentation/#/stages

Does that look like it can do what you're looking for?
